pkg/util/lora/lora.go

```diff
@@ -27,3 +33,27 @@ func AdjustedRSSI(channelRSSI, snr float32) float32 {
 	}
 	return rssi
 }
+
+// ReceivedAtImprovable describes an entity that holds RxMetadata and timestamp.
+type ReceivedAtImprovable interface {
```

Comment:
Nit: just leave it as `UplinkMessage` - since we're just looking for a shape, it's fine.